### PR TITLE
[55789] make_tensor_proto() now respects byte order of numpy input arrays

### PR DESCRIPTION
**Pull request by mjangstadt**
Fixes Issue: https://github.com/tensorflow/tensorflow/issues/55789

Previously make_tensor_proto() did not respect byte order of numpy input arrays as demonstrated in this repro: 
https://colab.research.google.com/gist/tilakrayal/cef07cf718456b6b07526ea38d73fb52/55789.ipynb

Now with these updates, the function takes into consideration the byte order of the system and input array and coerces the input array to match prior to utilizing it within the function.

### CODE REVIEW

File: tensorflow/python/framework/tensor_util.py [hunk 1]
```diff
@@ -520,7 +521,17 @@ def make_tensor_proto(values, dtype=None, shape=None, verify_shape=False,
     if nparray.size * nparray.itemsize >= (1 << 31):
       raise ValueError(
           "Cannot create a tensor proto whose content is larger than 2GB.")
-    tensor_proto.tensor_content = nparray.tobytes()
+    
+    #ensure sys and dtype endianness match
+    sys_is_le = sys.byteorder == 'little'
+    numpy_is_le = numpy_dtype.byteorder == '<'
```
Comment (edloper) [inline tensorflow/python/framework/tensor_util.py:527]: It looks like byteorder can also be "=" or '|'. 
[https://numpy.org/doc/stable/reference/generated/numpy.dtype.byteorder.html]

For '=' (native byte order) and '|' (NA -- e.g., 'i1' or 'S2' dtypes), we don't need to do any conversion.

File: tensorflow/python/framework/tensor_util.py [hunk 2]
```diff
@@ -520,7 +521,17 @@ def make_tensor_proto(values, dtype=None, shape=None, verify_shape=False,
...
+    if sys_is_le and not numpy_is_le:
+      tensor_proto.tensor_content = nparray.astype('<i2').tobytes()
```
Comment (edloper) [inline tensorflow/python/framework/tensor_util.py:529]: Why i2 (int16) and f4 (float32) here?

**Comment by gbaned** [comment]
@mjangstadt Can you please check @tomytsai1's comments and keep us posted ? Thank you!

**Comment by gbaned** [comment]
@mjangstadt Any update on this PR? Please. Thank you!

**Pull request closed without merging**